Recrutify/Recrutify.DataAccess/Repositories/ScheduleRepository.cs

```diff
@@ -20,6 +20,17 @@ public ScheduleRepository(IOptions<MongoSettings> options)
         public Task<List<Schedule>> GetByUserPrimarySkillAsync(IEnumerable<Guid> userIds, DateTime date, Guid primarySkillId)
         {
             var filter = _filterBuilder.In(u => u.UserId, userIds) & _filterBuilder.Eq(u => u.UserPrimarySkill.Id, primarySkillId);
+            return GetCollectionByDatePeriod(filter, date, 1);
+        }
+
+        public Task<List<Schedule>> GetByDateAsync(Guid userId, DateTime date)
+        {
+            var filter = _filterBuilder.Eq(u => u.UserId, userId);
```

Comment:
This method will always return one document. So the one option make your private method returning FindDefinition (without ToListAsync) and in your public methods call ToListAsync and FirstOrDefaultAsync. Or in this method call await and call FirstOrDefaultAsync for your list. I prefer first option

Reply:
+

Reply:
FirstOrDefaultAsync